prompting/utils/timer.py

```diff
@@ -0,0 +1,11 @@
+import time
+
+
+class Timer:
+    def __enter__(self):
+        self.start_time = time.time()
+        return self
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        self.end_time = time.time()
```

Comment:
Use perf_counter for benchmarks
```suggestion
        self.end_time = time.perf_counter()
```